diff -ru nano-2.2.6/configure.ac nano-2.2.6-tinfo/configure.ac
--- nano-2.2.6/configure.ac	2010-11-22 07:43:06.000000000 +0100
+++ nano-2.2.6-tinfo/configure.ac	2011-12-13 10:26:51.175153589 +0100
@@ -442,11 +442,11 @@
     AC_CHECK_HEADERS(ncurses.h)
 
     if test x$enable_utf8 != xno; then
-	AC_CHECK_LIB(ncursesw, get_wch, [CURSES_LIB="-lncursesw" CPPFLAGS="-I/usr/include/ncursesw $CPPFLAGS" CURSES_LIB_NAME=ncursesw CURSES_LIB_WIDE=yes])
+	AC_CHECK_LIB(ncursesw, get_wch, [CURSES_LIB="-lncursesw -ltinfo" CPPFLAGS="-I/usr/include/ncursesw $CPPFLAGS" CURSES_LIB_NAME=ncursesw CURSES_LIB_WIDE=yes])
     fi
 
     if eval "test x$CURSES_LIB_NAME = x"; then
-	AC_CHECK_LIB(ncurses, initscr, [CURSES_LIB="-lncurses" CURSES_LIB_NAME=ncurses])
+	AC_CHECK_LIB(ncurses, initscr, [CURSES_LIB="-lncurses -ltinfo" CURSES_LIB_NAME=ncurses])
     fi
 fi
 
